x/incentives/keeper/distribute_test.go

```diff
@@ -1376,3 +1450,174 @@ func (s *KeeperTestSuite) TestCreateGroupGaugeAndDistribute() {
 	}
 
 }
+
+// deepCopyGroupGauge creates a deep copy of the passed in group gauge.
+func deepCopyGroupGauge(src types.GroupGauge) types.GroupGauge {
+	gaugeRecords := make([]types.InternalGaugeRecord, len(src.InternalGaugeInfo.GaugeRecords))
+	for i, record := range src.InternalGaugeInfo.GaugeRecords {
+		gaugeRecords[i] = types.InternalGaugeRecord{
+			GaugeId:          record.GaugeId,
+			CurrentWeight:    record.CurrentWeight,
+			CumulativeWeight: record.CumulativeWeight,
+		}
+	}
+
+	return types.GroupGauge{
+		GroupGaugeId: src.GroupGaugeId,
+		InternalGaugeInfo: types.InternalGaugeInfo{
+			TotalWeight:  src.InternalGaugeInfo.TotalWeight,
+			GaugeRecords: gaugeRecords,
+		},
+		SplittingPolicy: src.SplittingPolicy,
+	}
+}
+
+// withUpdatedVolumes takes in a group gauge and a list of updated cumulative volumes (ordered) and updates the contents of the gauge to
+// reflect these new volumes.
+// It is only intended to be used to set expected values for test cases.
+func (s *KeeperTestSuite) withUpdatedVolumes(groupGauge types.GroupGauge, updatedCumulativeVolumes []osmomath.Int) types.GroupGauge {
+	// Ensure there aren't more volumes to update than records in group gauge
+	s.Require().True(len(updatedCumulativeVolumes) <= len(groupGauge.InternalGaugeInfo.GaugeRecords))
+
+	// We make a deep copy of the group gauge to ensure we don't modify the original input/defaults
+	updatedGroupGauge := deepCopyGroupGauge(groupGauge)
+
+	newTotalWeight := osmomath.ZeroInt()
+	for i, updatedVolume := range updatedCumulativeVolumes {
+		currentRecord := groupGauge.InternalGaugeInfo.GaugeRecords[i]
+		updatedRecord := types.InternalGaugeRecord{
+			GaugeId:          currentRecord.GaugeId,
+			CurrentWeight:    updatedVolume.Sub(currentRecord.CumulativeWeight),
+			CumulativeWeight: updatedVolume,
+		}
+		updatedGroupGauge.InternalGaugeInfo.GaugeRecords[i] = updatedRecord
+		newTotalWeight = newTotalWeight.Add(updatedRecord.CurrentWeight)
+	}
+	updatedGroupGauge.InternalGaugeInfo.TotalWeight = newTotalWeight
+
+	return updatedGroupGauge
+}
+
+// withSplittingPolicy returns a deep copy of the passed in group gauge with the splitting policy set to the passed in value.
+func withSplittingPolicy(groupGauge types.GroupGauge, splittingPolicy types.SplittingPolicy) types.GroupGauge {
+	// We make a deep copy of the group gauge to ensure we don't modify the original input/defaults
+	updatedGroupGauge := deepCopyGroupGauge(groupGauge)
+	updatedGroupGauge.SplittingPolicy = splittingPolicy
+
+	return updatedGroupGauge
+}
+
+func (s *KeeperTestSuite) TestSyncGroupGaugeWeights() {
```

Comment:
This is a good catch, thank you for flagging this. The tests were written before the `syncVolumeSplitGauge ` logic was abstracted into a helper, and in hindsight I should have refactored the tests to fit this.

Reply:
Cleaned this up in recent commits by doing the following:
1. Converted existing test to run on `syncVolumeSplitGauge`
2. Added new thin test that runs on `syncGroupGaugeWeights` with a single happy path (volume split policy) and a single error case (unsupported policy)

Note that I did not include `Evenly` policy as I believe this was added as a placeholder and should be removed instead of getting tied into more features